### PR TITLE
GTK3.0 Upgrade

### PR DESCRIPTION
**Pull request by anmol27katyani**
Please check the following
Anything i could have missed,
Please point it out.

### CODE REVIEW

**Comment by walterbender** [comment]
File "/home/walter/Activities/amazonas-activity/TurtleArtActivity.py", line 37, in <module>
    from sugar.activity import activity

You should be importing from sugar3

Did you test this?

**Comment by chimosky** [comment]
@octamois also see https://wiki.sugarlabs.org/go/Features/GTK3/Porting

File: TurtleArtActivity.py [hunk 1]
```diff
@@ -22,11 +22,12 @@
 #OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 #THE SOFTWARE.
 
-import pygtk
-pygtk.require('2.0')
-import gtk
+import gi
+gi.require_version('Gtk', '3.0')
+from gi.repository import Gtk
+from gi.repository import GObject
```
Comment (rhl-bthr) [inline TurtleArtActivity.py:28]: Gdk should also be imported
`from gi.repository import Gdk`

File: TurtleArtActivity.py [hunk 2]
```diff
@@ -191,7 +192,7 @@ def check_buttons_for_fit(self):
         self._toolbox.toolbar.remove(self.stop_button)
         self._view_toolbar.remove(self._coordinates_toolitem)
 
-        if gtk.gdk.screen_width() / 14 < style.GRID_CELL_SIZE:
+        if Gtk.Gdk.screen_width() / 14 < style.GRID_CELL_SIZE:
```
Comment (rhl-bthr) [inline TurtleArtActivity.py:195]: `Gdk` is independent of `Gtk` unlike in `pygtk2`
Also `Gdk` does not have a `screen_width` method. Instead you may use `Gdk.Screen.width()`

File: TurtleArtActivity.py [hunk 3]
```diff
@@ -237,7 +238,7 @@ def do_load_ta_project_cb(self, button, new=False):
             _logger.debug('setting watch cursor')
             if hasattr(self.get_window(), 'get_cursor'):
                 self._old_cursor = self.get_window().get_cursor()
-                self.get_window().set_cursor(gtk.gdk.Cursor(gtk.gdk.WATCH))
```
Comment (rhl-bthr) [inline TurtleArtActivity.py:240]: `Gdk` is independent of `Gtk` unlike in `pygtk2`

File: TurtleArtActivity.py [hunk 4]
```diff
@@ -259,7 +260,7 @@ def do_load_ta_plugin_cb(self, button):
         if hasattr(self, 'get_window'):
             if hasattr(self.get_window(), 'get_cursor'):
                 self._old_cursor = self.get_window().get_cursor()
-                self.get_window().set_cursor(gtk.gdk.Cursor(gtk.gdk.WATCH))
+                self.get_window().set_cursor(Gtk.Gdk.Cursor(Gtk.Gdk.WATCH))
```
Comment (rhl-bthr) [inline TurtleArtActivity.py:263]: `Gdk` is independent of `Gtk` unlike in `pygtk2`

File: turtleblocks.py [hunk 5]
```diff
@@ -878,7 +877,7 @@ def _fill_challenges_list(self, store):
         '''
         for filepath in self._scan_for_challenges():
             pixbuf = None
-            pixbuf = gtk.gdk.pixbuf_new_from_file_at_size(
+            pixbuf = Gtk.Gdk.pixbuf_new_from_file_at_size(
```
Comment (rhl-bthr) [inline turtleblocks.py:880]: You need to import `GdkPixbuf` and use `GdkPixbuf.Pixbuf.new_from_file_at_size`

**Review by rhl-bthr — CHANGES_REQUESTED** [review]
The changes requested are having multiple instances, of which, I have marked a few. Kindly make the changes in all the instances

**Comment by rhl-bthr** [comment]
@octamois, kindly test the patch after making the requested changes.

Also, mention the testing platform here, for reference

**Comment by anmol27katyani** [comment]
@Pro-Panda I'll get to these changes, just busy for a while.
Sorry for late reply.

**Pull request closed without merging**